workspaces/vscode-python/Dockerfile

```diff
@@ -0,0 +1,109 @@
+USER coder
+COPY requirements.txt /
+RUN export arch=`uname -m` \
+ && curl -sfLO https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-${arch}.sh \
+ && /bin/bash Miniconda3-latest-Linux-${arch}.sh -b -p /home/coder/miniconda \
+ && PATH="/home/coder/miniconda/bin:$PATH" \
+ && /home/coder/miniconda/condabin/conda init bash \
+ && conda config --add channels conda-forge \
+ && conda run -n base pip install -r /requirements.txt
+
+# Cleaning up the cache in the same step when we install is important if you
+# want to minimize the image size.
+USER coder
+RUN ( code-server --disable-telemetry --force \
+		# vscode support for python, including debugger 
+		--install-extension ms-python.python \
+		# auto-generate templates for function docstrings 
+		--install-extension njpwerner.autodocstring \
+		# auto-fix indentation for multiline contexts 
+		--install-extension KevinRose.vsc-python-indent \
+		# visualize indentation with color 
+        --install-extension oderwat.indent-rainbow \
+		# visualize csv files in vscode window
+		--install-extension mechatroner.rainbow-csv \
+		# linting for markdown files
+		--install-extension DavidAnson.vscode-markdownlint \
+		# format tables in markdown files
+        --install-extension fcrespo82.markdown-table-formatter \
+	) && \
+	rm -rf /home/coder/.local/share/code-server/CachedExtensionVSIXs
+
+# Prepare the entrypoints. The entrypoint.sh part is based on what code-server
+# originally had, so it may need to be updated when code-server is updated. We
+# avoid running fixuid if the user is root (for local testing in PL). We also
+# disable auth as PL handles it.
+USER root
+RUN rm -f /run/fixuid.ran && \
+	echo "" > /usr/bin/entrypoint.sh && chmod 0755 /usr/bin/entrypoint.sh && \
+	/usr/bin/env echo -e '#!/usr/bin/env bash \n' \
+	'set -eu \n' \
+	'# We do this first to ensure sudo works below when renaming the user. \n' \
+	'# Otherwise the current container UID may not exist in the passwd database. \n' \
+	'[ $(id -u) -ne 0 ] && eval "$(fixuid -q)" \n' \
+	' \n' \
+	'if [ "${DOCKER_USER-}" ]; then \n' \
+	'  USER="$DOCKER_USER" \n' \
+	'  if [ "$DOCKER_USER" != "$(whoami)" ]; then \n' \
+	'    echo "$DOCKER_USER ALL=(ALL) NOPASSWD:ALL" | sudo tee -a /etc/sudoers.d/nopasswd > /dev/null \n' \
+	'    # Unfortunately we cannot change $HOME as we cannot move any bind mounts \n' \
+	'    # nor can we bind mount $HOME into a new home as that requires a privileged container. \n' \
+	'    sudo usermod --login "$DOCKER_USER" coder \n' \
+	'    sudo groupmod -n "$DOCKER_USER" coder \n' \
+	' \n' \
+	'    sudo sed -i "/coder/d" /etc/sudoers.d/nopasswd \n' \
+	'  fi \n' \
+	'else \n' \
+	'  sudo rm /etc/sudoers.d/nopasswd \n' \	
+	'fi \n' \
+	'\n' \
+	'dumb-init /usr/bin/code-server --auth none "$@" \n' >> /usr/bin/entrypoint.sh
+RUN rm /requirements.txt
+
+EXPOSE 8080
+
+# TODO: Maybe customize the preferences to set the theme to auto-adjust to the
+# browser, turn off more telemetry, etc.
```

Comment:
Do you intend to resolve this TODO before merging? At least regarding telemetry, it seems like that's already quite disabled.

Reply:
This comment can probably just be deleted @Victorsss-Orz - @echuber2 is already tracking this in his future improvement of this.

Reply:
I'm going to commit a few minor changes to these comments. It's worth noting that, iirc, code-server has an extra option to turn off _their own telemetry_ in addition to the internal settings that disable VS Code's telemetry.

Reply:
Removed this comment in a commit below.